File: Server/src/main/java/com/github/dr/rwserver/util/log/exp/VariableException.java
package com.github.dr.rwserver.util.log.exp;

/**
 * @author Dr
 */
public class VariableException extends RuntimeException {
    public VariableException(String type) {
        super(com.github.dr.rwserver.util.log.ErrorCode.valueOf(type).getError());
    }

    public static class ArrayRuntimeException extends RuntimeException {
        public ArrayRuntimeException(String type) {
            super(type);
        }
    }

    public static class MapRuntimeException extends RuntimeException {
        public MapRuntimeException(String type) {
            super(type);
        }
    }
}

File: Server/src/main/java/com/github/dr/rwserver/util/log/exp/VariableException.kt
package com.github.dr.rwserver.util.log.exp

import com.github.dr.rwserver.util.log.ErrorCode
import java.lang.RuntimeException

/**
 * @author Dr
 */
class VariableException(type: String) : RuntimeException(ErrorCode.valueOf(type).error)

{
    class ArrayRuntimeException(type: String) : RuntimeException(ErrorCode.valueOf(type).error)

    class ObjectMapRuntimeException(info: String) : RuntimeException(info)

    class MapRuntimeException(type: String) : RuntimeException(ErrorCode.valueOf(type).error)
}